src/System.Net.Primitives/tests/FunctionalTests/SocketAddressTest.cs

```diff
@@ -75,10 +75,49 @@ public static void ToString_Expected_Success(AddressFamily family, int size, boo
             Assert.Equal(expected, sa.ToString());
         }
 
-        [ActiveIssue(37997, TestPlatforms.AnyUnix)]
-        [Theory] // recombine into ToString_Expected_Success when #37997 is fixed
-        [InlineData((AddressFamily)12345, -1, false, "12345:32:{0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0}")]
-        public static void ToString_UnknownFamily_Success(AddressFamily family, int size, bool fillBytes, string expected) =>
-            ToString_Expected_Success(family, size, fillBytes, expected);
+        [Theory]
+        [InlineData((AddressFamily)12345, -1)]
+        [InlineData((AddressFamily)12345, 16)]
+        public static void ToString_UnknownFamily_Throws(AddressFamily family, int size)
+        {
+            Assert.Throws<PlatformNotSupportedException>(() => size >= 0 ? new SocketAddress(family, size) : new SocketAddress(family));
+        }
+
+        [Theory]
+        [InlineData(AddressFamily.Packet)]
+        [InlineData(AddressFamily.Netlink)]
+        [InlineData(AddressFamily.ControllerAreaNetwork)]
+        [PlatformSpecific(~TestPlatforms.Linux)]
+        public static void ToString_UnsupportedFamily_Throws(AddressFamily family)
+        {
+            Assert.Throws<PlatformNotSupportedException>(() => new SocketAddress(family));
+        }
+
+        [Theory]
+        [InlineData(AddressFamily.Packet)]
+        [InlineData(AddressFamily.ControllerAreaNetwork)]
```

Comment:
What about Netlink?

Reply:
Netlink will throw even on Linux. We left it in enum but did not finish PAL implementation because of difficulties with ProtocolType. 

If we want to cover it with test, we would need to add one more Linux test or add linux handling to test expecting to throw.

Reply:
> We left it in enum but did not finish PAL implementation because of difficulties with ProtocolType.

So we added an enum value but it's completely unusable?  There's no way someone could be successful with it today as things stand?

Reply:
generally yes and that is also true for other existing values. I was hoping we would make progress on #14041 (which is AF agnostic) and then the value would make more sense. 
And we run out of time for finding solution for platform-independent ProtocolType for Netlink.

We could still fix it so one can create Netlink SocketAddress but that is not that useful imho without ability to crate and use Socket it self.

Reply:
I don't think we should be adding useless values. We should only add a new value when it's useful in at least some situation. It sounds like you're saying there is literally no way it could be useful to anyone without us making further changes, in which case we shouldn't ship it now.

Reply:
I'll take another look tomorrow. We can remove Netlink from enum and add it back when we have PAL (and perhaps other) changes ready. 
I assume it would not hold back this PR. 
We may need new issue so we have handle to push changes to 3.0 branch.